src/parser.ts

```diff
@@ -474,3 +474,10 @@ const getDefinitionsMemoized = memoize(getDefinitions)
 function hasDefinitions(schema: LinkedJSONSchema): schema is JSONSchemaWithDefinitions {
   return 'definitions' in schema
 }
+
+function definitionMatchesSchema(definition: LinkedJSONSchema, schema: LinkedJSONSchema): unknown {
+  if ('enum' in definition && 'enum' in schema) {
+    return definition.enum === schema.enum
+  }
+  return definition === schema
+}
```

Comment:
I'm not sold that this is a particularly good solution 🙃 it feels weird to hardcode the case for enums. But I'm not familiar enough with the codebase to have any deeper insights.

Reply:
I noticed the same on #466, but not related to enums. In their case, they had a property like this:

```
"status": {
  "$ref": "#/definitions/status_block",
  "description": "..."
}
```

See my [comment](https://github.com/bcherny/json-schema-to-typescript/issues/466#issuecomment-1200571984) as to why.

For a more universal fix that goes beyond just enums, I wonder about:
* copy the original parsed schema content
* cull certain fields from any subschemas that use `$ref`
* _(Now the $ref resolver can make the $ref referentially equal to what it's referencing, so that `json-schema-to-typescript` doesn't generate a new type (or inline, depending on what version you're using?))_
* During code generation, reference the original json for things like `description`, which shouldn't affect type relationships, but should certainly affect generated doc strings.

wdyt?

Reply:
Honestly, although that sounds reasonable, I'm very new to this project and haven't done a deep dive into it. So if you say that's a good fix then I believe you! 😄